File: diagflow/observability/report.py
"""Markdown report generation for diagnosis results."""

from __future__ import annotations

from datetime import datetime
from typing import Any

from diagflow.core.diag_agent import DiagnosisReport


def render_report(report: DiagnosisReport) -> str:
    """Render a diagnosis report as formatted Markdown."""
    lines = [
        f"# 🔍 诊断报告: {report.component}.{report.problem_type}",
        "",
        f"- **事件 ID**: {report.event_id}",
        f"- **组件**: {report.component}",
        f"- **问题类型**: {report.problem_type}",
        f"- **耗时**: {report.duration_ms:.0f}ms",
        f"- **匹配知识库**: {'✅' if report.matched_knowledge else '❌'}",
        f"- **时间**: {datetime.now().isoformat()}",
        "",
        "---",
        "",
        "## 📋 根因分析",
        "",
        f"**根因**: {report.root_cause}",
        "",
        f"**置信度**: {_confidence_badge(report.confidence)}",
        "",
        "---",
        "",
        "## 🔧 修复建议",
        "",
    ]

    for i, suggestion in enumerate(report.suggestions, 1):
        lines.append(f"{i}. {suggestion}")

    lines.extend([
        "",
        "---",
        "",
        "## 📊 证据链",
        "",
    ])

    if report.evidence_summary:
        for ev in report.evidence_summary:
            lines.append(f"- **[{ev.get('source_agent', 'unknown')}]** "
                         f"{ev.get('summary', '')} "
                         f"_(置信度: {ev.get('confidence', 0.5):.0%})_")
    else:
        lines.append("_(无证据收集)_")

    lines.extend([
        "",
        "---",
        "",
        "## 🔍 排查过程追溯",
        "",
        f"完整排查日志位于: `/tmp/diagflow/{report.event_id}/`",
        "",
    ])

    return "\n".join(lines)


def _confidence_badge(level: str) -> str:
    badges = {
        "high": "🟢 高",
        "medium": "🟡 中",
        "low": "🔴 低",
    }
    return badges.get(level, level)

File: internal/observability/report/report.go
// Package report mirrors diagflow/observability/report.py — Markdown rendering
// of a diagnosis report.
package report

import (
	"fmt"
	"strings"
	"time"

	"github.com/Admaing/diagflow/internal/diagagent"
)

// Render renders a diagnosis report as Markdown.
func Render(r *diagagent.Report) string {
	var b strings.Builder
	fmt.Fprintf(&b, "# 诊断报告: %s.%s\n\n", r.Component, r.ProblemType)
	fmt.Fprintf(&b, "- **事件 ID**: %s\n", r.EventID)
	fmt.Fprintf(&b, "- **组件**: %s\n", r.Component)
	fmt.Fprintf(&b, "- **问题类型**: %s\n", r.ProblemType)
	fmt.Fprintf(&b, "- **耗时**: %.0fms\n", r.DurationMS)
	matched := "❌"
	if r.MatchedKnowledge {
		matched = "✅"
	}
	fmt.Fprintf(&b, "- **匹配知识库**: %s\n", matched)
	fmt.Fprintf(&b, "- **时间**: %s\n\n", time.Now().Format(time.RFC3339))
	b.WriteString("---\n\n## 根因分析\n\n")
	fmt.Fprintf(&b, "**根因**: %s\n\n", r.RootCause)
	fmt.Fprintf(&b, "**置信度**: %s\n\n", confidenceBadge(r.Confidence))
	b.WriteString("---\n\n## 修复建议\n\n")
	for i, s := range r.Suggestions {
		fmt.Fprintf(&b, "%d. %s\n", i+1, s)
	}
	b.WriteString("\n---\n\n## 证据链\n\n")
	if len(r.EvidenceSummary) == 0 {
		b.WriteString("_(无证据收集)_\n\n")
	} else {
		for _, ev := range r.EvidenceSummary {
			source, _ := ev["source_agent"].(string)
			summary, _ := ev["summary"].(string)
			fmt.Fprintf(&b, "- **[%s]** %s\n", source, summary)
		}
	}
	b.WriteString("\n")
	return b.String()
}

func confidenceBadge(level string) string {
	switch level {
	case "high":
		return "🟢 高"
	case "medium":
		return "🟡 中"
	case "low":
		return "🔴 低"
	default:
		return level
	}
}
